podpac/core/coordinates/coordinates.py

```diff
@@ -580,15 +580,50 @@ def udims(self):
         dims
         """
 
-        return tuple(dim for c in self._coords.values() for dim in c.dims)
+        return tuple(dim for c in self._coords.values() for dim in c.udims)
+
+    @property
+    def shape(self):
+        """:tuple: Tuple of the number of coordinates in each dimension."""
+        
+        return tuple(size for c in self._coords.values() for size in c.shape)
+
+    @property
+    def ndim(self):
+        """:int: Number of dimensions. """
+
+        return len(self.shape)
+
+    @property
+    def size(self):
+        """:int: Total number of coordinates."""
+
+        if len(self.shape) == 0:
+            return 0
+        return np.prod(self.shape)
+
+    @property
+    def bounds(self):
+        """:dict: Dictionary of (low, high) coordinates bounds in each unstacked dimension"""
+        return {dim: self[dim].bounds for dim in self.udims}
+
+    @property
+    def area_bounds(self):
+        """:dict: Dictionary of (low, high) coordinates area_bounds in each unstacked dimension"""
+        return {dim: self[dim].area_bounds for dim in self.udims}
 
     @property
     def coords(self):
         """
         :xarray.core.coordinates.DataArrayCoordinates: xarray coords, a dictionary-like container of coordinate arrays.
         """
 
-        x = xr.DataArray(np.empty(self.shape), coords=[c.coordinates for c in self._coords.values()], dims=self.dims)
+        coords = OrderedDict()
+        for c in self._coords.values():
+            coords.update(c.coords)
+        # TODO just return coords?

```

Comment:
I think its good to return the DataArray coords object, unless they provide a simpler means for constructing their coordinates. It shouldn't cost too much and it ensures we are always keeping up to date with xarray changes